### ISSUE TITLE
Reimport can cause internal server error

### ISSUE THREAD

**Issue by madchap** [DefectDojo/django-DefectDojo#3327]
```
ERROR:django.request:Internal Server Error: /api/v2/reimport-scan/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/mixins.py", line 19, in create
    self.perform_create(serializer)
  File "./dojo/api_v2/views.py", line 1081, in perform_create
    serializer.save(push_to_jira=push_to_jira)
  File "./dojo/api_v2/serializers.py", line 1330, in save
    finding.component_name = finding.component_name if finding.component_name else component_name
UnboundLocalError: local variable 'component_name' referenced before assignment
```

**Comment by valentijnscholten** [comment]
which commit are you on? this trace is different from current dev.

**Comment by valentijnscholten** [comment]
actually I see where the issue is.

**Comment by valentijnscholten** [comment]
fixed now in dev and 1.10.0.
